taxonomy.html

```diff
@@ -0,0 +1,478 @@
+          taxonomy and taxon name. Depending on the taxonomy the object may
+          implement other interfaces derived from <code>Taxon</code> interface.
+          For example, the returned object for the <code>role</code> taxonomy
+          implements <code>RoleTaxon</code> interface.
+        </p>
+        <p>
+          The example logs out all parents of the <code>bibliography</code>
+          role.
+        </p>
+        <pre class="example highlight">
+function logParents(taxon) {
+  for (let parent of taxon.parents) {
+    console.log(parent.name);
+    console.log(':');
+    logParents(parent);
+    console.log(';');
+  }
+}
+let taxon = document.a11ement.taxonOf('role', 'bibliography');
+logParents(taxon);
+</pre>
+      </section>
+      <section>
+        <h3>
+          <code>import</code> method
+        </h3>
+        <p>
+          Imports taxonomies from the given library into the document
+          taxonomies.
+        </p>
+        <p>
+          In the example bunch of ARIA DPUB roles are imported into document's
+          taxonomy.
+        </p>
+        <pre class="example highlight">
+var DPUBRoles = {
+  appendix: {
+    description: "appendix",
+    parents: "section",
+    attributes: {
+      landmark: "appendix"
+    }
+  },
+  bibliography: {
+    description: "bibliography",
+    parents: "list",
+    attributes: {
+      landmark: "bibliography"
+    }
+    owns: "biblioentry"
+  },
+  biblioentry: {
+    description: "reference",
+    parents: "listitem"
+  }
+};
+
+document.a11ement.import({
+  id: 'ARIA:DPUB',
+  roles: DPUBRoles
+});
+</pre>
+        <p>
+          The method can be used to extend existing taxonomies. If, for
+          example, the library describes an role that exists in the taxonomy
+          then the role taxon is extended by new properties, if they don't
+          conflict to the existing ones. In latter case an exception is thrown.
+        </p>
+        <p>
+          In the example the root role taxon is extended by new <a>states</a>
+          and <a>relations</a>. These states and relations will be applicable
+          to all roles in the taxonomy.
+        </p>
+        <pre class="example highlight">
+var library = {
+  roles: {
+    '*': {
+      states: [
+        'highlighable', 'highlighted'
+      ],
+      relations: [
+        'hihglight-comment-for',
+        'hihglight-commented-by'
+      ]
+    }
+  },
+  states: {
+    highlighable: { },
+    highlighted: {
+      description: 'user choice'
+    }
+  },
+  relations: {
+    'hihglight-comment-for': { },
+    'hihglight-commented-by': {
+      description: 'commented by'
+    }
+  }
+}
+</pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        TaxoLibrary dictionary
+      </h2>
+      <p>
+        A collection of taxonomies extensions to the document's taxonomies.
+      </p>
+      <pre class="idl">
+dictionary TaxoLibrary {
+  DOMString id;
+  object roles;
```

Comment:
I'm not sure, but `object` might not be the right type here. You might need different dictionary types of each of these. Like:
``` 
{
   AccessibleState states;
}

dictionary AccessibleState {
   DOMString state;
   DOMString foo;
}
```